torchvision/prototype/transforms/_utils.py

```diff
@@ -34,3 +34,15 @@ def get_image_dimensions(image: Union[PIL.Image.Image, torch.Tensor, features.Im
     else:
         raise TypeError(f"unable to get image dimensions from object of type {type(image).__name__}")
     return channels, height, width
+
+
+def _extract_types(sample: Any) -> Iterator[Type]:
+    return query_recursively(lambda id, input: type(input), sample)
+
+
+def has_any(sample: Any, *types: Type) -> bool:
+    return any(issubclass(type, types) for type in _extract_types(sample))
+
+
+def has_all(sample: Any, *types: Type) -> bool:
+    return not bool(set(types) - set(_extract_types(sample)))
```

Comment:
@pmeier maybe we could code `has_all` and `has_any` using torch tree_flatten and simplify the code:
```python
from typing import Any, Type

from torch.utils._pytree import tree_flatten


def has_any(sample: Any, *types: Type) -> bool:
    flat_sample, _ = tree_flatten(sample)
    return any(issubclass(type(obj), types) for obj in flat_sample)


def has_all(sample: Any, *types: Type) -> bool:
    flat_sample, _ = tree_flatten(sample)
    return not bool(set(types) - set([type(obj) for obj in flat_sample]))
```